llvm/lib/Support/CommandLine.cpp

```diff
@@ -2060,6 +2062,24 @@ bool parser<float>::parse(Option &O, StringRef ArgName, StringRef Arg,
   return false;
 }
 
+// parser<DenseSet<StringRef> implementation
+//
+void parser<DenseSet<StringRef>>::printOptionDiff(
+    const Option &O, const DenseSet<StringRef> &V,
+    OptionValue<DenseSet<StringRef>> D, size_t GlobalWidth) const {}
+
+bool parser<DenseSet<StringRef>>::parse(Option &O, StringRef ArgName,
+                                        StringRef Arg,
```

Comment:
Are you certain that the contents of `StringRef Arg` outlives the DenseSet<StringRef> returned? All the others above return the contents by value.

I would prefer a cl::opt<std::string> which is parsed and validated in the ICP pass. Is there an advantage to your proposed approach and are there any precedents in the codebase? IMO this option feels overly restrictive when rejecting duplicated strings from the user.